Enterprise-CMCS/managed-care-review
# Rate Refactor: Add contracts and rates tables

## Summary

This PR adds new Contract and Rate tables with their associated revisions and DB functions for updating them both and fetching a coherent list of revisions for each. 

Things that are not done yet:
* Errors are all generic, not an easy way to tell the difference between a random prisma error and passing an invalid ID into one of the db functions
* DB actions aren't in transactions yet, so it is possible to get the db in a bad state by doing things on top of one another
* The findContract/findRate functions chunk revisions based on timestamps. In reality each of those timestamps represent a single revision being submitted. Right now if you submit two revisions at too close to the same time it there isn't enough precision to tell them apart in those functions. That should be fixable but it's where it's at right now. This is why the tests I have here have sleeps in them between submissions. 
* Right now Drafts aren't really findable, they are omitted from findContract/findRate

#### Related issues

https://qmacbis.atlassian.net/browse/MR-3234

#### Test cases covered

There are two tests files for now mostly just testing happy path scenarios. 

## QA guidance

Please check this out and add a test! Try different combinations of submitting and updating and unlocking rates and contracts and make sure that the list of revisions spat out by findContract and findRate make sense.
